src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -61,7 +63,7 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   start(): void {
-    if (this.isClosed()) {
+    if (this.isClosed() && !this.isConnecting) {
```

Comment:
Could you please explain why do we need the `isConnecting` here, is `start` getting called multiple times unintendedly?

Reply:
It's needed for an existing unit test that detects that start isn't called twice.  With the async behavior in connect now the isClosed existing method is using the underlying WebSocket connection state which may or may not reflect open by the time the second call is made so we need an isConnecting state.

Reply:
Yup I thought so, thanks for explaining.

Reply:
Yup I thought so, thanks for explaining.